sugondat/shim/src/sugondat_rpc/mod.rs

```diff
@@ -107,33 +108,41 @@ impl Client {
     async fn get_header_and_extrinsics(
         &self,
         block_hash: Option<[u8; 32]>,
-    ) -> anyhow::Result<(Header, Vec<sugondat_subxt::ExtrinsicDetails>)> {
+    ) -> Result<(Header, Vec<sugondat_subxt::ExtrinsicDetails>), subxt::error::Error> {
         let block_hash = block_hash.map(H256::from);
+
+        let conn = self.connector.ensure_connected().await;
+        let res = match block_hash {
+            Some(h) => conn.subxt.blocks().at(h).await,
+            None => conn.subxt.blocks().at_latest().await,
+        }?;
+
+        let header = res.header();
+        let body = res
+            .extrinsics()
+            .await?
+            .iter()
+            .collect::<Result<Vec<_>, _>>()?;
+        Ok((header.clone(), body))
+    }
+
+    async fn wait_header_and_extrinsics(
```

Comment:
bikeshed: probably `await`

would be nice to add a docs line explaining that it waits for a block with the given hash to be available.

Reply:
I added the docs line! Regarding the naming, I simply adapted it to match the existing function `wait_finalized_height`. If you believe it would be better to update all three names to `await`, I can easily do that!

Reply:
for a follow-up (#208)